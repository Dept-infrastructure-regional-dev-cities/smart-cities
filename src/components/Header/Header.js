import React from 'react';
import PropTypes from 'prop-types';
import CategoryNavContainer from '../CategoryNavContainer/CategoryNavContainer';

import style from './Header.scss';

const Header = props => (
  <header className={style.header}>
    <div className={style.header_govau_container}>
      <span>*STAR*</span>
      <strong>GOV.AU</strong>
      <span>Official Australian Government Website</span>
    </div>
    <div className={style.header_project_status_container}>
      <div className={style.header_project_status}>BETA</div>
    </div>
    <h1 className={style.title}>Header</h1>

    <CategoryNavContainer
      // pass in categoryId so the nav updates when the category changes
      categoryId={props.categoryId}
      cityId={props.cityId}
    />
  </header>
);

Header.propTypes = {
  cityId: PropTypes.string.isRequired,
  categoryId: PropTypes.string.isRequired,
};

export default Header;
