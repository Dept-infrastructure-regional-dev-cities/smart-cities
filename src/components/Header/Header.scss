@import '~style/variables';

header {
  background: $color-primary--500;
  color: $color-white;

  a {
    color: $color-white;
    text-decoration: underline;
    text-decoration-color: $color-highlight;
  }
}

.govContainer {
  display: flex;
  height: 48px;
  padding-left: 14px;
  flex-flow: column wrap;
  align-items: flex-start;
  align-content: flex-start;
  justify-content: center;

  @include for-desktop-up {
    padding-left: 30px;
    height: 32px;
    align-items: center;
    flex-flow: row nowrap;
    justify-content: flex-start;
    background: $color-primary--400;

    > * {
      margin-right: 8px;
    }
  }
}

.compactTitle,
.compactSubtitle {
  @include for-desktop-up {
    display: none;
  }
}

.govAu,
.govWebsite {
  display: none;

  @include for-desktop-up {
    display: inline;
  }
}

.govAu {
  font-size: 14px;
  font-weight: 500;
}

.govWebsite {
  font-size: 12px;
  font-weight: 100;
}

.mainContainer {
  display: none;

  @include for-desktop-up {
    display: flex;
    width: 100%;
    justify-content: space-between;
  }
}

.titleContainer,
.orgContainer {
  padding: 0 32px;
  max-height: 176px;
  display: inline-flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: 24px;
}

.star {
  height: 22px;
  width: 22px;
  max-width: 22px;
  flex: 0 0 100%;
  margin-right: 5px;

  @include for-desktop-up {
    height: 18px;
    width: 18px;
    flex: 0 1 auto;
  }
}

.projectStatus {
  margin-top: 24px;
  text-align: center;
  width: 80px;
  font-weight: 600;
  border: 1px solid $color-badge;
  border-radius: 17px;
  background: $color-badge;
}

.title {
  font-size: 36px;
  font-weight: 300;
  line-height: 46px;
  margin-top: 10px;
}

.compactTitle {
  font-size: 14px;
  font-weight: 600;
  flex: 0 0 16px;
}

.subtitle {
  font-size: 12px;
  text-transform: uppercase;
}

.compactSubtitle {
  text-transform: uppercase;
  font-weight: 200;
  font-size: 10px;
  flex: 0 0 14px;
}

.orgLink {
  margin-top: 14px;
}

.orgLogo {
  height: 65px;
  width: 102px;
}

.contactLink {
  font-size: 22px;
}
