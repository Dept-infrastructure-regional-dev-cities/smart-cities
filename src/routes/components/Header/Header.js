import React from 'react';

import style from './Header.scss';

const Header = () => {
  return (
    <header className={style.Header}>
      <h1>Header</h1>
    </header>
  )
};

export default Header;
