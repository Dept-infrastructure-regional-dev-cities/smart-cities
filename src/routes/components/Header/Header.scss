@import '~style/helper';

.wrapper {
    border: 1px solid blue;
}
