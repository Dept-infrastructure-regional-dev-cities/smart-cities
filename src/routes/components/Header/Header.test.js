import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

it('should match Snapshot', () => {
  const renderedValue =  renderer.create(
    <Header openNav={() => {}}/>
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
