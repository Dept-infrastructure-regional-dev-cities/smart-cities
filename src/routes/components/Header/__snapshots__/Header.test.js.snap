// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<header
  className={undefined}
>
  <h1>
    Header
  </h1>
</header>
`;
